src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,104 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class Storage {
+    protected String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads task list from file.
+     *
+     * @return List of tasks loaded from file.
+     * @throws FileNotFoundException If file does not exist.
+     */
+    public List<Task> load() throws FileNotFoundException {
+        List<Task> tasks = new ArrayList<>();
+
+        // check if file exists
+        File f = new File(filePath);
+
+        if (f.exists()) {
+            Scanner s = new Scanner(f);
+            Task task;
```

Comment:
Variables should be initialized where they are declared and they should be declared in the smallest scope possible.